_posts/2018-02-16-Java-Time2.md

```diff
@@ -0,0 +1,1564 @@
+        } catch (RestException e) {
+            e.printStackTrace();
+        }
+        
+        return template;
+    }
+    
+    public List<DomainTemplate> fetchAllTemplatesForEnterprise(Enterprise enterprise) {
+        DomainTemplatesFetcher fetcher = enterprise.getDomainTemplates(); // 9
+        List<DomainTemplate> list = new ArrayList<>();
+
+        try {
+            if (fetcher.count() > 0) { // 10
+                list = fetcher.get(); // 11
+            }
+        } catch (RestException e) {
+            e.printStackTrace();
+        }
+
+        return list;
+    }
+    
+    public void deleteTemplate(DomainTemplate template) { // 12
+        if (template == null) {
+            System.out.println("Template does not exist");
+        } else {
+            System.out.println("Template " + template.getName() + " will be deleted");
+            try {
+                template.delete(); // 13
+            } catch (RestException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+}
+{% endhighlight %}
+
+In Line 1 of the `verifyTemplate` method, we perform a search of an existing Enterprise object for a VSD Domain Template object that matches the name for our domain template that is passed in via `NetworkDescriptor`. If the Domain Template already exists, then we return that Domain Template object.  If the Domain Template does not exist, then in Line 2 we call the `createTemplate` method to create and return a new Domain Template object.
+
+Line 3 of the `fetchTemplate` method sets up a search filter based on the name of our domain template.  In Line 4, we create an instance of `DomainTemplatesFetcher`.  Then Line 5 retrieves the Domain Template object matching the name in the search filter. 
+
+In Line 6 of the `createTemplate` method, we call the constructor of the VSPK DomainTemplate class to create a brand new Domain Template object and then in Line 7 we assign our template name.  Finally, in Line 8, we attach our new Domain Template object to the already existing Enterprise object.
+
+Lines 9 to 13 of methods `fetchAllTemplatesForEnterprise` and `deleteTemplate` will be covered later when we discuss how to remove our network layout from the VSD.
+
+# Adding the Domain Template to our Network
+
+Here is the updated version of the `NetworkBuilder` class that integrates with `NetworkDomainTemplateHandler`.  There are 2 minor changes as shown below.  At Line 1, we instantiate the new handler and at Line 2, we call its `verifyTemplate` method with our existing Enterprise object and the `NetworkDescriptor` object which contains the name of our domain template.
+
+{% highlight java %}
+import java.util.List;
+import net.nuagenetworks.bambou.RestException;
+import net.nuagenetworks.vspk.v5_0.DomainTemplate;
+import net.nuagenetworks.vspk.v5_0.Enterprise;
+import net.nuagenetworks.vspk.v5_0.VSDSession;
+
+public class NetworkBuilder {
+    private String url = "https://135.228.4.108:8443";

```

Comment:
Don't use a 135.... IP in documentation, use one from https://tools.ietf.org/html/rfc5737

Suggestion: `198.51.100.108`